core/src/main/java/org/elasticsearch/monitor/process/ProcessProbe.java

```diff
@@ -19,12 +19,249 @@
 
 package org.elasticsearch.monitor.process;
 
-/**
- *
- */
-public interface ProcessProbe {
+import org.elasticsearch.bootstrap.Bootstrap;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.management.ThreadMXBean;
+import java.lang.reflect.Method;
+
+import static org.elasticsearch.monitor.jvm.JvmInfo.jvmInfo;
+
+public class ProcessProbe {
+
+    private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
+    private static final ThreadMXBean threadMxBean = ManagementFactory.getThreadMXBean();
+
+    private static final Method getMaxFileDescriptorCountField;
+    private static final Method getOpenFileDescriptorCountField;
+    private static final Method getProcessCpuLoad;
+    private static final Method getProcessCpuTime;
+    private static final Method getCommittedVirtualMemorySize;
+
+    static {
+        Method method = null;
+        try {
+            method = osMxBean.getClass().getDeclaredMethod("getMaxFileDescriptorCount");
+            method.setAccessible(true);
+        } catch (Exception e) {
+            // not available
+        }
+        getMaxFileDescriptorCountField = method;
```

Comment:
similar comment from my OS level review, extract it to a method to make sure we don't leak the method down the code